central/clusterinit/backend/certificate/certprovider.go

```diff
@@ -34,6 +35,14 @@ func (c *certProviderImpl) GetBundle() (clusters.CertBundle, uuid.UUID, error) {
 	return certBundle, id, nil
 }
 
+func (c *certProviderImpl) GetCRSCert() (*mtls.IssuedCert, uuid.UUID, error) {
+	cert, id, err := clusters.IssueSecuredClusterCRSCertificates()
+	if err != nil {
+		return nil, uuid.Nil, errors.Wrap(err, "generating CRS certificate")
```

Comment:
Well, this is surely a matter of style. I am trying to be very consistent about this actually.

So when I wrapping an error for

```
err := foo()
if err != nil {
    return err
}
```

I am essentially always doing it like this:
```
err := foo()
if err != nil {
    return errors.Wrap(err, "fooing")
}
```

I don't see the need to repeat the "failed" on every level while bubbling up, as it reads much better (in my opinion) without it. In an error message it is clear that something failed (this should be indicated already by the error at the bottom). Now, while wrapping I am trying to express *during which operation* something failed.
Does this make sense?

Reply:
> Well, this is surely a matter of style.

I guess you can call it style in a very broad sense... but to me this is more about user experience.

> I don't see the need to repeat the "failed" on every level while bubbling up, as it reads much better (in my opinion) without it. In an error message it is clear that something failed (this should be indicated already by the error at the bottom). Now, while wrapping I am trying to express _during which operation_ something failed. Does this make sense?

I do see your point. The argument that it reduces method length and avoids repetition on each level does make sense.

However what concerns me more, is that in some cases a concise description of an operation might be too short for an uninitiated user to understand. I had seen cases (not sure whether in stackrox or elsewhere) where it was not clear to me whether the proverbial "fooing" was actually something that was supposed to be happening, or a sign of trouble in itself. The additional "error when..." or "... failed" makes it clear that we wanted something to happen but it did not. I don't have an example handy unfortunately.

IMHO think errors like:
```
failed to frobnicate dependencies: fooing failed: i/o timeout
```
read more naturally, especially when you read `:` as "because"
than:
```
frobnicating dependencies: fooing: i/o timeout
```

I took a very quick look at what we do elsewhere in the codebase, and out of ~3k uses of Wrap, over 700 mention `failed` or `error`. So it's not completely uncommon...

It would be nice if we had an established guidelines in this regard 🤔 

But anyway I'm certainly not going to block on this in case I didn't manage to convince you 😉

Reply:
Ok, fine, I would also hope for some established guidelines here.
Don't feel like arguing about this one. :-)